### PR TITLE
oguri: reduce supported platforms

### PR DESCRIPTION
**Pull request by viraptor**
###### Description of changes

This doesn't work on darwin which has no wayland support.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by viraptor** [comment]
Since this is for wayland compositors, there are no other platforms to include here, right?

**Comment by AndersonTorres** [comment]
Usually I would `inherit (wayland.meta) platforms;`, however recently it bugged ofborg because God knows why wayland included Darwin support.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/wayland/oguri/default.nix [hunk 1]
```diff
@@ -36,6 +36,6 @@ stdenv.mkDerivation rec {
     description = "A very nice animated wallpaper daemon for Wayland compositors";
     license = licenses.mit;
     maintainers = with maintainers; [ AndersonTorres ];
-    platforms = platforms.unix;
+    platforms = platforms.linux;
```
[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/tools/wayland/oguri/default.nix:39]: We might want to wait for #214906 for this to work.

**Comment by alyssais** [comment]
> (have I said why I hate the linux-only approach of wayland?)

Wayland works on FreeBSD, and is not far off from NetBSD and OpenBSD.  I have a Nixpkgs branch that can build Sway for FreeBSD, but I'm still in the process of upstreaming it.

**Pull request merged**